napari/layers/points/tests/test_points.py

```diff
@@ -105,9 +105,14 @@ def test_selecting_points():
     np.random.seed(0)
     data = 20 * np.random.random(shape)
     layer = Points(data)
+    layer.mode = 'select'
     layer.selected_data = [0, 1]
     assert layer.selected_data == [0, 1]
 
+    # test switching to 3D
+    layer.dims.ndisplay = 3
+    assert layer.selected_data == []
```

Comment:
@kevinyamauchi I don't understand why the selected data is lost, rather than just the box not being drawn?

Reply:
It's because the way our modes are currently implemented, the selected points are cleared when switching from the `select` to another mode and when going from 2D to 3D , the mode is automatically switched to the `pan_zoom` mode (since selection isn't supported in 3D for now).

That being said, it's fairly straightforward to allow point selection in 3D from the command line/API (as requested in #901). I just thought would be better to make a small PR here to fix the bug that causes a crash and then make the larger PR as a follow-up.

Reply:
approach sounds fine for now, but agreed a larger followup sounds good too